.github/workflows/cypress_tests.yml

```diff
@@ -54,51 +53,52 @@ jobs:
         env:
           PORT: 5000
 
-      - name: Install pnpm
-        uses: pnpm/action-setup@v2
+      - uses: pnpm/action-setup@v2
         with:
           version: latest
-
-      - name: Run and test app
+      - name: Run Webapp
+        working-directory: packages/webapp
+        run: |
+          pnpm install --config.auto-install-peers=true --no-frozen-lockfile --force
+          pnpm dev &
         env:
           VITE_WEATHER_API_KEY: ${{secrets.VITE_WEATHER_API_KEY}}
           VITE_ENV: development
           VITE_GOOGLE_OAUTH_CLIENT_ID: ${{secrets.VITE_GOOGLE_OAUTH_CLIENT_ID}}
           VITE_DO_BUCKET_NAME: litefarm
           NODE_ENV: development
           VITE_API_URL: http://localhost:5000
-          CYPRESS_RECORD_KEY: "2630f414-4914-48b0-907d-aa3a9cc6a30b"
-        uses: cypress-io/github-action@v4.2.0
+          CYPRESS_COVERAGE: true
+
+      - name: Run E2E test (English)
+        continue-on-error: true
+        env:
+          CYPRESS_RECORD_KEY: ${{secrets.CYPRESS_RECORD_KEY}}
+          CYPRESS_USER: 0
+        uses: cypress-io/github-action@v6
         with:
-          working-directory: packages/webapp
-          install-command: pnpm install --config.auto-install-peers=true --no-frozen-lockfile --force
-          install: true
-          start: pnpm dev
-          wait-on: "http://localhost:3000"
-          wait-on-timeout: 120
+          working-directory: packages/end-to-end
           headed: true
           browser: chrome
           record: true
           parallel: true
           group: "UI - Chrome"
-          spec: cypress/e2e/happyPath.spec.js
+
+      - name: Archive backend logs
+        uses: actions/upload-artifact@v3
+        with:
+          name: logs
+          path: packages/api/logs
 
       - name: set code coverage badge
-        run: npx -p check-code-coverage update-badge --from '${GITHUB_WORKSPACE}/packages/webapp/coverage/coverage-summary.json'
+        working-directory: "."
+        run: npx -p check-code-coverage update-badge --from '${GITHUB_WORKSPACE}/packages/end-to-end/coverage/coverage-summary.json'
         env:
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

Comment:
Looks like we don't need to set it up it should automatically be created. 

https://docs.github.com/en/actions/security-guides/automatic-token-authentication

I am not sure if it is failing for me because of my permissions on GitHub. Will try on my own fork.